File: 2427-number-of-common-factors/2427-number-of-common-factors.java
class Solution {
    public int commonFactors(int a, int b) {
        int count = 1;
        int min = Math.min(a,b);
        
        for(int i = 2; i <= min; i++){
            if(a % i == 0 && b % i == 0) count++;
        }
        return count;
    }
}

File: 2427-number-of-common-factors/2427-number-of-common-factors.swift
class Solution {
    func commonFactors(_ a: Int, _ b: Int) -> Int {
        let min = min(a,b)
        var cnt = 1
        var i = 2
        
        while i <= min {
            if (a % i == 0 && b % i == 0) {
                cnt += 1
            }
            i += 1
        }
        
        return cnt
    }
}